frame/executive/src/lib.rs

```diff
@@ -101,6 +98,13 @@ pub type CheckedOf<E, C> = <E as Checkable<C>>::Checked;
 pub type CallOf<E, C> = <CheckedOf<E, C> as Applyable>::Call;
 pub type OriginOf<E, C> = <CallOf<E, C> as Dispatchable>::Origin;
 
+/// The last time the runtime upgrade happened.
+///
+/// Stores the the `spec_version` as [`LastRuntimeUpgrade`] of this runtime.
+const LAST_RUNTIME_UPGRADE: &'static [u8] = b":last_runtime_upgrade:";
+/// The type used to store the `spec_version` under [`LAST_RUNTIME_UPGRADE`].
+type LastRuntimeUpgrade = Compact<u32>;
```

Comment:
`spec_name` may reasonably change, so should probably include it (or a hash of it) here.

Reply:
`set_code` currently prevents from changing the spec name. When would it happen that the spec name changes?